### DISCUSSION TITLE
Firebase plugins could make Google Play Services available

### DISCUSSION THREAD

**Discussion by kroikie** [firebase/flutterfire#5553] [Feature request]
From the [docs](https://firebase.google.com/docs/cloud-messaging/android/client#sample-play):

> Apps that rely on the Play Services SDK should always check the device for a compatible Google Play services APK before accessing Google Play services features. It is recommended to do this in two places: in the main activity's onCreate() method, and in its onResume() method. The check in onCreate() ensures that the app can't be used without a successful check. The check in onResume() ensures that if the user returns to the running app through some other means, such as through the back button, the check is still performed.
> 
> If the device doesn't have a compatible version of Google Play services, your app can call `GoogleApiAvailability.makeGooglePlayServicesAvailable()` to allow users to download Google Play services from the Play Store.

The google_sign_in plugin already [ensures](https://github.com/flutter/plugins/blob/master/packages/google_sign_in/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java#L458) Google Play Services is present and I think the Firebase Android plugins should as well -- they could attach lifecycle listeners to the activity to check for `onCreate` and `onResume` automatically.  (We could also expose a Dart API for it, but I'd rather not since it's Android specific.)

I don't think there's an easy way for plugins to prevent the Flutter app from starting up but the plugins could queue up method calls until the error is resolved.

This issue is based on a conversation with @lukef although I think Posse has already worked around it.

**Comment by kroikie** [comment]
@collinjackson

The issue at https://github.com/flutter/flutter/issues/11237 has been closed and moved here. Future collaboration on this issue will be done here.

**Comment by markusaksli-nc** [comment]
Just mentioning that this also means that authenticating in Firebase Auth using 
Google Play Games https://firebase.google.com/docs/auth/android/play-games is a missing native functionality.
